### PR TITLE
muiv5 - Breaking Apart - Core/Layout Changes

### PR DESCRIPTION
**Pull request by jdslaugh**
### Description
- Breaking apart https://github.com/lyft/clutch/pull/2288
- Contains core/layout changes

### Testing Performed
manual

### CODE REVIEW

**Review by smonero — APPROVED** [review]

File: frontend/packages/core/src/AppLayout/drawer.tsx [hunk 1]
```diff
@@ -5,9 +5,9 @@ import {
   Avatar as MuiAvatar,
   Drawer as MuiDrawer,
   List,
-  ListItem,
+  ListItemButton,
   Typography,
-} from "@material-ui/core";
+} from "@mui/material";
```
Comment (scarlettperry) [inline frontend/packages/core/src/AppLayout/drawer.tsx:10]: should we update the package.json in the PR as well? when running `make frontend-lint` I'm seeing a few of:

`@mui/material' should be listed in the project's dependencies. Run 'npm i -S @mui/material' to add it  import/no-extraneous-dependencies`

& 

`@mui/icons-material' should be listed in the project's dependencies. Run 'npm i -S @mui/icons-material' to add it  import/no-extraneous-dependencies`

File: frontend/packages/core/src/AppLayout/search.tsx [hunk 2]
```diff
@@ -198,12 +198,16 @@ const SearchField: React.FC = () => {
...
 
-  const renderResult = (option: SearchIndex) => {
+  const renderResult = (props, option: SearchIndex) => {
     const handleSelection = () => {
       navigate(option.path);
     };
 
-    return <Result option={option} handleSelection={handleSelection} />;
+    return (
+      <li {...props}>
```
Comment (scarlettperry) [inline frontend/packages/core/src/AppLayout/search.tsx:207]: for my understanding, why is the `<li>` element needed?

File: frontend/packages/core/src/AppProvider/themes.tsx [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
+  DeprecatedThemeOptions,
+  StyledEngineProvider,
+  ThemeProvider,
+} from "@mui/material";
+import { createTheme, PaletteOptions, useTheme as useMuiTheme } from "@mui/material/styles";
+import { StylesProvider } from "@mui/styles";
+
+declare module "@mui/styles/defaultTheme" {
+  interface DefaultTheme extends MuiTheme {}
+}
```
Comment (scarlettperry) [inline frontend/packages/core/src/AppProvider/themes.tsx:15]: locally these lines are throwing errors for me - `Cannot find module` & `Invalid module name in augmentation`. Curious if you're seeing this as well or if it's a local issue

File: frontend/packages/core/src/AppLayout/user.tsx [hunk 4]
```diff
@@ -48,7 +48,7 @@ const Paper = styled(MuiPaper)({
 
 const Popper = styled(MuiPopper)({
   padding: "0 12px",
-  marginLeft: "12px",
+  offset: "12px",
```
Comment (scarlettperry) [inline frontend/packages/core/src/AppLayout/user.tsx:51]: for my understanding, why offset vs marginLeft?

**Comment by scarlettperry** [comment]
can you share deets on the short link state changes that are in the PR? noticed it seems to undo what was merged in https://github.com/lyft/clutch/pull/2294 so just curious as I'm not familiar with the feature

File: frontend/packages/core/src/AppLayout/drawer.tsx (hunk 1 above)
Comment (jdslaugh) [reply frontend/packages/core/src/AppLayout/drawer.tsx:10]: They should all be updated, might just be the branch. I've removed all of @material-ui in the `matv5` branch.

File: frontend/packages/core/src/AppLayout/search.tsx (hunk 2 above)
Comment (jdslaugh) [reply frontend/packages/core/src/AppLayout/search.tsx:207]: https://mui.com/material-ui/migration/v5-component-changes/#update-renderoption

It used to be wrapped around the render options by default but they removed it in muiv5, this puts it back to keep parity for the moment.

File: frontend/packages/core/src/AppLayout/user.tsx (hunk 4 above)
Comment (jdslaugh) [reply frontend/packages/core/src/AppLayout/user.tsx:51]: https://mui.com/material-ui/migration/v5-component-changes/#popper
https://popper.js.org/docs/v2/migration-guide/#4-remove-all-css-margins

The popper api was upgraded to v2 and margin was removed, has to be replaced with offset

File: frontend/packages/core/src/AppProvider/themes.tsx (hunk 3 above)
Comment (jdslaugh) [reply frontend/packages/core/src/AppProvider/themes.tsx:15]: Can't say I've seen those issues, are you building this branch or the big one?

**Comment by jdslaugh** [comment]
> can you share deets on the short link state changes that are in the PR? noticed it seems to undo what was merged in #2294 so just curious as I'm not familiar with the feature

That would be because when I built the initial branch the feature wasn't present, then I broke them apart and the copy+paste was missing it. Let me fix that.

File: frontend/packages/core/src/AppProvider/themes.tsx (hunk 3 above)
Comment (scarlettperry) [reply frontend/packages/core/src/AppProvider/themes.tsx:15]: oh yeah this was when I was just trying to build this branch, building the big branch works for me.

File: frontend/packages/core/src/AppLayout/drawer.tsx (hunk 1 above)
Comment (scarlettperry) [reply frontend/packages/core/src/AppLayout/drawer.tsx:10]: sg! yeah this was before i realized that the big branch had all the import updates

**Review by scarlettperry — APPROVED** [review]
lgtm!

**Pull request merged**